gui/default/syncthing/core/syncthingController.js

```diff
@@ -1734,6 +1769,33 @@ angular.module('syncthing.core')
             });
         };
 
+        $scope.showRemoteNeed = function (device) {
+            var promises = [];
+            $scope.remoteNeedDevice = device.deviceID;
+            var i = 0;
+            $scope.deviceFolders(device).forEach(function(folder) {
```

Comment:
I think forEach has a hack, where by if you do `function(folder, idx)` it should give you the index.

Reply:
Yes, but I don't need the index of all that I iterate over, only of those that aren't complete.